### ISSUE TITLE
Using SwinFusion

### ISSUE THREAD

**Issue by Adam11072000** [Linfeng-Tang/SwinFusion#33]
Hey guys,

I tried out with your models and i see that the pre-trained models all receive a single channel of input.
My main goal is to use your model with ImageNet and IR masks and produce an RGB picture.
how can i do that?

here is a snapshot of what i did:
![image](https://github.com/user-attachments/assets/9cf16962-feda-4f38-bf19-3ba4c43558e4)

**Comment by Linfeng-Tang** [comment]
Can you provide me with a visible image and its infrared counterpart?

Adam1107 ***@***.***> 于2024年7月21日周日 00:09写道：

**Comment by Adam11072000** [comment]
well, my goal is to use the image, and an infrared mask, not nessecarily the same image with a counterpart.
the IR part shape is a (3, H, W)  and the RGB shape is (3, H, W) too.
what we see here in the error above is that the weights of the trained model (at least the first layer), expects a single channel of input, whereas the code gives the ability to exploit RGB.

**Comment by Adam11072000** [comment]
i accessed the weights when loading the pretrained model, and the shape of the first layer is (30, 1 3, 3), where 30 is the output channels, 1 is the input channels (this guy right here should be 3).

**Comment by ccatian** [comment]
> i accessed the weights when loading the pretrained model, and the shape of the first layer is (30, 1 3, 3), where 30 is the output channels, 1 is the input channels (this guy right here should be 3).

Hi, have you resolved the issue of fusing two RGB color images to generate a color image at the end?
